docs/workflow.md

```diff
@@ -538,35 +488,36 @@ workflow {
     )
 
     publish:
-    ch_samples >> 'samples'
+    samples = ch_samples
 }
 
 output {
     samples {
-        path { sample -> "fastq/${sample.id}" }
+        path { sample -> "fastq/${sample.id}/" }
     }
 }
 ```
 
 The above example will publish each channel value to a different subdirectory. In this case, each pair of FASTQ files will be published to a subdirectory based on the sample ID.
 
-The closure can even define a different path for each individual file by returning an inner closure, similar to the `saveAs` option of the {ref}`publishDir <process-publishdir>` directive:
+The closure can even define a different path for each individual file using the `>>` operator:
 
 ```nextflow
 output {
     samples {
         path { sample ->
-            { filename -> "fastq/${sample.id}/${filename}" }
+            sample.fastq_1 >> "fastq/${sample.id}/"
+            sample.fastq_2 >> "fastq/${sample.id}/"
         }
     }
 }
 ```
 
-The inner closure will be applied to each file in the channel value, in this case `sample.fastq_1` and `sample.fastq_2`.
+Each `>>` specifies a *source file* and *publish target*. The source file should be a file or collection of files, and the publish target should be a directory or file name. If the publish target ends with a slash, it is treated as the directory in which source files are published. Otherwise, it is treated as the target filename of a source file. Only files that are published with the `>>` operator are saved to the output directory.
```

Comment:
The publish target was not the output e.g. `samples` in this example? this looks to me more a "publish directory" or just "output directory"

Reply:
I have changed the meaning of publish target in this iteration to match the overall syntactic/semantic changes

What was previously a "publish target" like `samples` is now simply an "output" or "output declaration".

Publish target now refers to the right-hand side of a publish statement e.g. `sample.fastq_1 >> 'fastq/'`

Essentially I have moved the concept of "publish target" into the path directive.

Basically every real-world use case I've seen when talking to users requires this fine-grained level of publishing, because of how they like to organize their output directory

Reply:
Not sure it still makes sense to talk about "target" in this case. May be just saying "directory" it can result more clear. Otherwise I'm OK

Reply:
The right-hand side can refer to a file name or directory name, that is why I call it "target" instead of "directory"